src/main/java/org/terasology/physicalstats/component/PhysicalStatsModifyEffect.java

```diff
@@ -19,55 +19,89 @@
 import org.terasology.engine.Time;
 import org.terasology.entitySystem.entity.EntityRef;
 import org.terasology.logic.delay.DelayManager;
-import org.terasology.logic.health.HealthComponent;
-import org.terasology.math.TeraMath;
 import org.terasology.physicalstats.event.OnPhysicalStatsModifierAddedEvent;
 import org.terasology.physicalstats.event.OnPhysicalStatsModifierRemovedEvent;
 
+/**
+ * Class that handles the application and removal of permanent or temporary physical stats modifiers to/from an entity.
+ */
 public class PhysicalStatsModifyEffect implements IPhysicalStatsModifyEffect {
+    /**  Used for getting the current time. */
     private final Time time;
+
+    /** Reference to the delay manager for adding temporary modifiers. */
     private final DelayManager delayManager;
 
+    /**
+     * Create an instance of this class using the passed in Context to get the Time and DelayManager instances.
+     * @param context
+     */
     public PhysicalStatsModifyEffect(Context context) {
         this.time = context.get(Time.class);
         this.delayManager = context.get(DelayManager.class);
     }
 
+    /**
+     * {@inheritDoc}
```

Comment:
Gotcha. This was made when I getting to grips on determining how to inherit docs from parent classes.